Added ability to change frame colour

ajlomagno opened the pull request:
Fixes #262

This pr adds the ability for the user to choose their own colour for the hoverzoom frame. So far the new option in the options screen is only in english.



Choosing a new colour:
![part1](https://user-images.githubusercontent.com/16764878/34456972-f7bd78c0-ed71-11e7-9ebd-52f227f2feb7.gif)

The new colour in action:
![part2](https://user-images.githubusercontent.com/16764878/34456971-f7a4adc2-ed71-11e7-8625-20c9d94fd8ae.gif)

extesy commented on js/common.js:
```diff
@@ -34,6 +34,7 @@ function loadOptions() {
     options.ambilightEnabled = options.hasOwnProperty('ambilightEnabled') ? options.ambilightEnabled : false;
     options.disabledPlugins = options.hasOwnProperty('disabledPlugins') ? options.disabledPlugins : [];
     options.centerImages = options.hasOwnProperty('centerImages') ? options.centerImages : false;
+    options.frameBackgroundColour = options.hasOwnProperty('frameBackgroundColour') ? options.frameBackgroundColour : "#ffffff"; 
```
Can you please follow American spelling conventions (colour -> color)?

extesy replied:
And also make this change in all other files as well.

extesy commented on js/hoverzoom.js:
```diff
@@ -124,7 +123,7 @@ var hoverZoom = {
                 position = {top:mousePos.top, left:mousePos.left};
             }
-
+                
```
please remove whitespace on this line

extesy commented on js/options.js:
```diff
@@ -143,8 +144,16 @@ function restoreOptions() {
+    $('#pickerFrameBackgroundColour').val(options.frameBackgroundColour);
     $('#selectCaptionLocation').val(options.captionLocation);
 
+    if(options.frameBackgroundColour == ""){
```
please add spaces before `(` and after `)`

extesy commented on js/options.js:
```diff
@@ -143,8 +144,16 @@ function restoreOptions() {
 
+    if(options.frameBackgroundColour == ""){
+        initColourPicker('#ffffff');
+    }else{
```
please add spaces around `else`

extesy commented on js/options.js:
```diff
@@ -143,8 +144,16 @@ function restoreOptions() {
+        initColourPicker(options.frameBackgroundColour);
+    }
+
+
```
remove extra empty line

extesy commented on js/options.js:
```diff
@@ -287,6 +296,20 @@ function populatePluginsTable() {
+        }
+    })
+
+    //colorPicker.attr('value', colorPicker.spectrum('get'));
```
remove this commented out line and previous empty line

extesy commented on js/options.js:
```diff
@@ -287,6 +296,20 @@ function populatePluginsTable() {
+    var colorPicker = $('#pickerFrameBackgroundColour').spectrum({
+        color: colour,
+        preferredFormat: "hex",
+        chooseText: "Select",
```
this needs to be localized (and `cancelText` as well)

extesy commented on js/options.js:
```diff
@@ -287,6 +296,20 @@ function populatePluginsTable() {
+        preferredFormat: "hex",
+        chooseText: "Select",
+        cancelText: "Cancel",
+        change:  function(color){
```
remove extra space after `:` and add a space before `{`

ajlomagno commented:
No problem! I'll take a look at what you wrote and fix up the pr. I originally intended to spell colour the American way but I guess I subconsciously wrote it the way I'm used to instead.

ajlomagno commented:
I fixed the issues mentioned above and added a simple color contrast implementation so the caption text color will change depending on the chosen frame color to keep the caption readable.
